sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -189,8 +204,17 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
 
   override def createDatabase(
       dbDefinition: CatalogDatabase,
-      ignoreIfExists: Boolean): Unit = withClient {
+      ignoreIfExists: Boolean): Unit = withClientWrappingException {
     client.createDatabase(dbDefinition, ignoreIfExists)
+  } { exception =>
+    if (exception.getClass.getName.equals(
+          "org.apache.hadoop.hive.metastore.api.AlreadyExistsException")
+        && exception.getMessage.contains(
+          s"Database ${dbDefinition.name} already exists")) {
+      Some(new DatabaseAlreadyExistsException(dbDefinition.name))
```

Comment:
Can we test it in the `VersionsSuite`, by updating the existing `$version: createDatabase` test? We need to make sure this wrap exception logic works for all hive versions.

Reply:
Updated the test. Thanks!